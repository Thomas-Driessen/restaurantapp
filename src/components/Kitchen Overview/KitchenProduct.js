import React from 'react';
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import AddIcon from '@material-ui/icons/Add';
import Grid from '@material-ui/core/Grid';


class KitchenProduct extends React.Component {

    render() {
        return (
            <div>
                { this.props.product ? (
                    <div>
                        <Card>
                            <CardContent>
                                <Typography gutterBottom variant="inherit" component="h2">
                                    {this.props.product.title}
                                </Typography>
                            </CardContent>
                            <CardActions>
                                <Grid container alignItems="flex-start" justify="flex-end" direction="row">
                                    <div style={{ display: 'flex', alignItems: 'right' }}>
                                        <IconButton aria-label="push" onClick={() => this.props.goToNext(this.props.product.title)}>
                                            <AddIcon />
                                        </IconButton>
                                    </div>
                                </Grid>
                            </CardActions>
                        </Card>
                    </div>
                ) : null}
            </div>
        )
    }
}
export default KitchenProduct