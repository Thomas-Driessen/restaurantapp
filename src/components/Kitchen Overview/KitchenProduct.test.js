import React from 'react';
import { render } from '@testing-library/react';
import KitchenProduct from './KitchenProduct';
import renderer from 'react-test-renderer';

const product = {
    title: ["Coca-Cola"],
    tableNumber: 3,
    state: [false],
    timestamp: "14:51:58"
};

it('KitchenProduct renders correctly', () => {
  const tree = renderer
    .create(<KitchenProduct />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render KitchenProduct', () => {
  const { getByText } = render(<KitchenProduct product={"Coca-Cola"} enabled={false} element={product} position={0}/>);
  const linkElement = getByText(/Coca-cola/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})