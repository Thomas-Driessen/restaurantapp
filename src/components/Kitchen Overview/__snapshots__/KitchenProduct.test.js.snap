// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KitchenProduct renders correctly 1`] = `<div />`;
